src/main/java/com/squad8/s8orangebackend/service/ProjectService.java

```diff
@@ -68,12 +67,12 @@ public Project fromDto(ProjectDto projectDto) {
         project.setUser(userService.getCurrentUser());
         return projectRepository.save(project);
     }
-
     public void deleteProject(Long id) {
+        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
boolean isProjectSaved = projectRepository.existsById(id);
        boolean isProjectUserOwner = Objects.equals(projectRepository.getReferenceById(id).getUser().getId(), userService.getCurrentUser().getId());
        if (isProjectSaved && isProjectUserOwner) {
            projectRepository.deleteById(id);
        }else {
            throw new UnauthorizedAccessException("User is not the owner of the project");
        }

Sugestão de código

Reply:
Feito como sugerido e criado a UnauthorizedAccessException no arquivo de exceptions do service.